lib/container-mapping.js

```diff
@@ -85,56 +86,77 @@ class ContainerMapping {
                 console.log(`PreJobStartTime not defined, using now-10secs`);
             }
             console.log(`PreJobStartTime: ${startTime}`);
+            let dockerVersion = [];
             let reportData = {
-                dockerVer: "",
-                dockerEvents: "",
-                dockerImages: ""
+                dockerVersion: "",
+                dockerEvents: [],
+                dockerImages: []
             };
-            yield exec.exec('docker --version', null, {
-                listeners: {
-                    stdout: (data) => {
-                        reportData.dockerVer = reportData.dockerVer.concat(data.toString());
-                    }
-                }
+            yield this.execCommand('docker --version', dockerVersion)
+                .catch((error) => {
+                throw new Error("Unable to get docker version: " + error);
```

Comment:
Should this really throw an error? What if the customer action doesn't use docker?

Reply:
We only capture images pushed using docker for now, this is a pre-req.

Reply:
Right, but if the customer runs an action that has nothing to do with building containers, the docker command would not be available, in which case this would throw an error, even though no true error exists.